### PR TITLE
Feature upload all pictures at once (#103)

### PR DESCRIPTION
**Pull request by mahdiatallah**
## Description

This pull request directly addresses issue #103.

The primary changes in this PR involve improving the handling of image uploads within the maintenance request process. Specifically, photos are now cached locally when creating a maintenance request and uploaded in bulk once all photos are added. This adjustment enhances the performance and stability of the process without altering the user flow.

## Changes Made
- **Caching of images**: Photos are now stored locally when selected, rather than being uploaded immediately. They are then uploaded all at once after selection.
- **UI message update**: The `CapturedMediaScreen` now displays the message `"Photo added to maintenance request!"` when a photo is added to provide clearer feedback to users.

  

- **New tests**: Added new tests specifically for verifying the caching functionality to ensure images are stored and retrieved correctly.

## Impact on User Flow
The user flow remains unchanged. This PR only includes an under-the-hood update to the caching and uploading process, and the sole visible change for users is the updated message in the UI when they add a photo.

---

*P.S.* `expo-file-system` was used to cache images. Here’s the link to the [documentation](https://docs.expo.dev/versions/latest/sdk/filesystem/).

**Testing:** For testing (beyond unit tests for caching and previous unit/e2e tests), one useful manual test is to create a maintenance request and add pictures to it. Each picture should only be uploaded to Firestore when all images are uploaded together, rather than immediately upon being added.

### CODE REVIEW

**Review by Rcc999 — CHANGES_REQUESTED** [review]
The tests cover the main functionalities, but I noticed some potential edge cases are missing. For example, what happens if the cacheImage function receives a corrupted blob or an empty string as picId? Adding tests for these edge cases would ensure robustness.

The logic for caching and retrieving images (cacheImage, getPictureBlob) is now reused in multiple places, which is great. However, have you considered moving some repeated operations, like blob uploads, into a utility function? This could further improve reusability and readability

The overall structure of the new caching and upload logic is well thought. Separating caching into its own utility functions (cacheImage, getPictureBlob) improves flexibility and keeps the components clean and focused on their responsibilities. And also, great job on including comprehensive tests for the new caching and retrieval logic. They cover various scenarios, including success and failure cases, and demonstrates a clear understanding of testing best practices.

File: app/utils/pictureCache.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  }
+};
+
+
+/**
+ * Deletes multiple pictures from the pictures directory
+ * @param {string[]} picIds - Array of picture IDs to delete
+ * @returns {Promise<void>}
+ */
+export const clearPictures = async (picIds: string[]): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (mahdiatallah) [reply app/utils/pictureCache.ts:97]: Good point about ensuring the list isn’t empty—that’s something I overlooked!

As for performance, it’s reasonable to assume tenants won’t upload too many pictures with their maintenance requests, so I think this approach should be efficient enough.

File: app/screens/issues_tenant/ReportScreen.tsx [hunk 2]
```diff
@@ -63,12 +72,32 @@ export default function ReportScreen() {
   const handleSubmit = async () => {
     setLoading(true); // Set loading to true when starting the submission
 
-    const tenantId = await getTenant(auth.currentUser?.uid || '');
-    console.log('url list for the pictures : ' , pictureList)
+
+    const tenantId = await getTenant("7OUEUmR82bCxt5Xgbli7");//auth.currentUser?.uid || '');
```
[the remark this answers is not in the document]
Comment (mahdiatallah) [reply app/screens/issues_tenant/ReportScreen.tsx:76]: Oh, I didn’t realize I had pushed that—sorry about that! It was just for testing purposes. I’ll remove it right away.

**Comment by mahdiatallah** [comment]
Thanks for the insight! I also realized I didn’t include documentation on `expo-file-system`, which could be helpful for future reference—especially to ensure our app can function offline. I’ll add that as a *P.S.*

I’ll also include steps for testing, since even though you knew how to test, it could be valuable for future reference (e.g., if someone modifies the caching mechanism, they’ll need to ensure it still works).

File: app/utils/pictureCache.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  }
+}
+
+/**
+ * Saves a blob to the pictures directory with the specified ID
+ * @param {Blob} blob - The blob to save
+ * @param {string} picId - The picture ID to use in the filename
+ * @returns {Promise<string>} The URI of the saved file
+ */
+export const cacheImage = async (blob: Blob, picId: string): Promise<string> => {
```
[the remark this answers is not in the document]
Comment (mahdiatallah) [reply app/utils/pictureCache.ts:24]: Will do, thanks! (And yes, it’s definitely better for generalizing to other file types, as you mentioned in your other comment.)
